### PR TITLE
SUS-1994 | ChatWidget - fallback to user registration date if he has no contributions yet

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-1994

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatWidget.class.php#L36](https://github.com/Wikia/app/blob/c35581b7055293486d12085714e2a606c1c44439/extensions/wikia/Chat2/ChatWidget.class.php#L36): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L62](https://github.com/Wikia/app/blob/c35581b7055293486d12085714e2a606c1c44439/extensions/wikia/Chat2/ChatWidget.class.php#L62): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L100](https://github.com/Wikia/app/blob/c35581b7055293486d12085714e2a606c1c44439/extensions/wikia/Chat2/ChatWidget.class.php#L100): Remove the unused function parameter "$input". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L100](https://github.com/Wikia/app/blob/c35581b7055293486d12085714e2a606c1c44439/extensions/wikia/Chat2/ChatWidget.class.php#L100): Remove the unused function parameter "$args". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MINOR][MINOR] [ChatWidget.class.php#L71](https://github.com/Wikia/app/blob/c35581b7055293486d12085714e2a606c1c44439/extensions/wikia/Chat2/ChatWidget.class.php#L71): Immediately return this expression instead of assigning it to the temporary variable "$vars". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatWidget.class.php#L36](https://github.com/Wikia/app/blob/5b3ed54a3ad4c70b00aad8211cae81759d303afb/extensions/wikia/Chat2/ChatWidget.class.php#L36): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L62](https://github.com/Wikia/app/blob/5b3ed54a3ad4c70b00aad8211cae81759d303afb/extensions/wikia/Chat2/ChatWidget.class.php#L62): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L98](https://github.com/Wikia/app/blob/5b3ed54a3ad4c70b00aad8211cae81759d303afb/extensions/wikia/Chat2/ChatWidget.class.php#L98): Remove the unused function parameter "$input". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L98](https://github.com/Wikia/app/blob/5b3ed54a3ad4c70b00aad8211cae81759d303afb/extensions/wikia/Chat2/ChatWidget.class.php#L98): Remove the unused function parameter "$args". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by sqreek — APPROVED** [review]

File: extensions/wikia/Chat2/ChatWidget.class.php [hunk 1]
```diff
@@ -192,23 +189,19 @@ function () use ( $userName ) {
 				// get stats for edit count and member since
 				$user = User::newFromName( $userName );
 
-				if ( $user instanceof User ) {
+				if ( $user instanceof User && $user->isLoggedIn() ) {
 					$userStatsService = new UserStatsService( $user->getId() );
```
Comment (ghost) [inline extensions/wikia/Chat2/ChatWidget.class.php]: I don't like the name `isLoggedIn()` as it suggests `isCurrentlyLoggedIn()` while it actually is `! isAnon()`. How about using `! isAnon()` which suggests exactly what it says?

**Review by ghost — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChatWidget.class.php#L36](https://github.com/Wikia/app/blob/7d5e7d48619438001876c522e428b791e5c0b8a0/extensions/wikia/Chat2/ChatWidget.class.php#L36): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L62](https://github.com/Wikia/app/blob/7d5e7d48619438001876c522e428b791e5c0b8a0/extensions/wikia/Chat2/ChatWidget.class.php#L62): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L98](https://github.com/Wikia/app/blob/7d5e7d48619438001876c522e428b791e5c0b8a0/extensions/wikia/Chat2/ChatWidget.class.php#L98): Remove the unused function parameter "$input". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [ChatWidget.class.php#L98](https://github.com/Wikia/app/blob/7d5e7d48619438001876c522e428b791e5c0b8a0/extensions/wikia/Chat2/ChatWidget.class.php#L98): Remove the unused function parameter "$args". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/Chat2/ChatWidget.class.php (hunk 1 above)
Comment (macbre) [reply extensions/wikia/Chat2/ChatWidget.class.php]: Changed in 7d5e7d48619438001876c522e428b791e5c0b8a0

**Review by ghost — APPROVED** [review]

**Pull request merged**